### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by nalimilan**
Improve docs for `fit`, `lm`, `glm` and `negbin`. Mention the two supported forms, document arguments and give them more explicit names. Repeat docstring so that users don't have to look at the `fit` docstring`.

Fix broken links to unexported functions. Stop including internal functions in the manual. Add list of included links to the `Link` docstring.

### CODE REVIEW

File: src/glmfit.jl [hunk 1]
```diff
@@ -421,33 +421,47 @@ function StatsBase.fit!(m::AbstractGLM,
     end
 end
 
+const FIT_GLM_DOC = """
+    In the first method, `formula` must be a `Formula` object and `data` a table
+    (in the Tables.jl definition, e.g. a data frame).
```
Comment (kleinschmidt) [inline src/glmfit.jl:426]:
```suggestion
    In the first method, `formula` must be a [StatsModels.jl `Formula` object](https://juliastats.org/StatsModels.jl/stable/formula/)
    and `data` a table (in the Tables.jl definition, e.g. a data frame).
```

Might not be clear to users what the formula object means... (also elsewhere with the same language)

**Review by kleinschmidt — COMMENTED** [review]
These changes look good.  Can you also update https://github.com/JuliaStats/GLM.jl/blob/master/docs/src/manual.md#fitting-glm-models to be consistent with the updated docstrings from the fit methods?  It also might be good to say explicitly that there are two interfaces, and have separate subsections describing them in docs/src/manual.md...put the matrix/vector one first, with examples, and then the formula one...

**Comment by nalimilan** [comment]
Woops, I forgot this PR again. I've pushed a commit which should address your comments. A larger refactoring can be done later.

File: docs/src/manual.md [hunk 2]
```diff
@@ -12,9 +12,17 @@ The [RDatasets package](https://github.com/johnmyleswhite/RDatasets.jl) is usefu
 
 ## Fitting GLM models
 
-To fit a Generalized Linear Model (GLM), use the function, `glm(formula, data, family, link)`, where,
-- `formula`: uses column symbols from the DataFrame data, for example, if `names(data)=[:Y,:X1,:X2]`, then a valid formula is `@formula(Y ~ X1 + X2)`
-- `data`: a DataFrame which may contain NA values, any rows with NA values are ignored
+Two methods can be used to fit a Generalized Linear Model (GLM):
+`glm(formula, data, family, link)` and `glm(X, y, family, link)`.
```
Comment (nalimilan) [inline docs/src/manual.md:16]: I hadn't realized that the order was `X`, `y`. This is completely counter-intuitive to me. :-/

Comment (kleinschmidt) [reply docs/src/manual.md:16]: Yeah I agree it's counter intuitive.  It would also be hugely breaking to change so...

File: src/lm.jl [hunk 3]
```diff
@@ -122,6 +122,37 @@ function StatsBase.fit!(obj::LinearModel)
     return obj
 end
 
+const FIT_LM_DOC = """
+    In the first method, `formula` must be a
+    [StatsModels.jl `Formula` object](https://juliastats.org/StatsModels.jl/stable/formula/)
+    and `data` a table (in the Tables.jl definition, e.g. a data frame).
```
Comment (kleinschmidt) [inline src/lm.jl:128]:
```suggestion
    and `data` a table (in the [Tables.jl](https://tables.juliadata.org/stable/) definition, e.g. a data frame).
```

File: src/glmfit.jl [hunk 4]
```diff
@@ -421,33 +421,48 @@ function StatsBase.fit!(m::AbstractGLM,
     end
 end
 
+const FIT_GLM_DOC = """
+    In the first method, `formula` must be a
+    [StatsModels.jl `Formula` object](https://juliastats.org/StatsModels.jl/stable/formula/)
+    and `data` a table (in the Tables.jl definition, e.g. a data frame).
```
Comment (kleinschmidt) [inline src/glmfit.jl:427]:
```suggestion
    and `data` a table (in the [Tables.jl](https://tables.juliadata.org/stable/) definition, e.g. a data frame).
```

File: src/negbinfit.jl [hunk 5]
```diff
@@ -39,20 +39,24 @@ function mle_for_θ(y::AbstractVector, μ::AbstractVector, wts::AbstractVector;
...
+    negbin(X::AbstractMatrix, y::AbstractVector, [link::Link];
+           <keyword arguments>)
 
 Fit a negative binomial generalized linear model to data, while simultaneously
 estimating the shape parameter θ. Extra arguments and keyword arguments will be
 passed to [`glm`](@ref).
 
+In the first method, `formula` must be a
+[StatsModels.jl `Formula` object](https://juliastats.org/StatsModels.jl/stable/formula/)
+and `data` a table (in the Tables.jl definition, e.g. a data frame).
```
Comment (kleinschmidt) [inline src/negbinfit.jl:53]:
```suggestion
and `data` a table (in the [Tables.jl](https://tables.juliadata.org/stable/) definition, e.g. a data frame).
```

**Review by kleinschmidt — COMMENTED** [review]
While we're at it, can you get rid of the uninformative landing page for the docs (that just says "linear and generalized linear models in Julia"; https://juliastats.org/GLM.jl/dev/) and make the manual.md page the main docs page? 

Otherwise, LGTM.

File: docs/src/manual.md (hunk 2 above)
Comment (palday) [reply docs/src/manual.md]: I think this was done to match the argument order in Python's StatsModels and a lot of "rawer" R packages.

Comment (nalimilan) [reply docs/src/manual.md]: That explains it. :-)

**Pull request merged**